File: TechNode.Core/Entities/OrderAggregate/Order.cs
namespace TechNode.Core.Entities.OrderAggregate;

public class Order
{
    public int Id { get; set; }
    
    public DateTime OrderCreated { get; set; }

    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;

    public required string BuyerEmail { get; set; }

    public PaymentSummary PaymentSummary { get; set; } = null!;

    public ICollection<OrderItem> OrderItems { get; set; } = [];

    public ShippingAddress ShippingAddress { get; set; } = null!;

    public DeliveryMethod DeliveryMethod { get; set; } = null!;

    public decimal Subtotal { get; set; }
    
    public required string PaymentIntendId { get; set; }

    public decimal GetTotal()
    {
        return Subtotal + DeliveryMethod.Price;
    }
}

File: client/src/app/shared/models/order.ts
export interface Order {
  id: number
  orderCreated: string
  orderStatus: string
  buyerEmail: string
  paymentSummary: PaymentSummary
  orderItems: OrderItem[]
  shippingAddress: ShippingAddress
  deliveryMethodId: number
  subtotal: number,
  total: number,
  shippingPrice: number,
  paymentIntendId: string
}

export interface PaymentSummary {
  last4: number
  brand: string
  expMonth: number
  year: number
}

export interface OrderItem {
  productId: number
  productName: string
  pictureUrl: string
  quantity: number
  price: number
}

export interface ShippingAddress {
  name: string
  line1: string
  line2: string | null
  city: string
  state: string
  postalCode: string
  country: string
}

export interface OrderToCreate{
  cartId: string,
  deliveryMethodId: number,
  paymentSummary: PaymentSummary,
  shippingAddress: ShippingAddress
}
